### PR TITLE
Un-focusing window during popup timer should not cause a popup to appear

### LINKED ISSUE #48634: Window unfocus doesn't stop tooltips from appearing
**Godot version:**

3.3

**Issue description:**

When you unfocus Godot window, the tooltip may still appear if the mouse was over tooltipable node when unfocusing.
MdyKKJkw9N

**Steps to reproduce:**
1. Move cursor over some editor button that has a tooltip
2. Before tooltip appears, Alt+Tab to another window
3. The tooltip will still appear even though the button is technically not hovered anymore

You can increase tooltip appear time to make it easier to observe.

Could be easily fixed by stopping the tooltip timer on notification unfocused. Also unfocusing should probably kill existing tooltips.

### PR DESCRIPTION
**Pull request by greatmomo**
Should fix #48634

Simply checks if `window->has_focus()` before showing a tooltip.

The issue also says "unfocusing should probably kill existing tooltips.", but as far as I saw during my testing, this already happens.

### CODE REVIEW

File: scene/main/viewport.cpp [hunk 1]
```diff
@@ -1618,7 +1618,8 @@ void Viewport::_gui_show_tooltip() {
 	gui.tooltip_popup->set_position(r.position);
 	gui.tooltip_popup->set_size(r.size);
 
-	gui.tooltip_popup->show();
+	if (window->has_focus())
+		gui.tooltip_popup->show();
```
Comment (Calinou) [inline scene/main/viewport.cpp:1622]: Braces are now required for single-line `if` statements:

```suggestion
	if (window->has_focus()) {
		gui.tooltip_popup->show();
	}

```

**Review by Calinou — DISMISSED** [review]

**Comment by Calinou** [comment]
Note that this will require a different pull request to fix this in the `3.x` branch, as the DisplayServer abstraction and Window class are not present there.

**Comment by akien-mga** [comment]
The two commits should be squashed, see [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase) for instructions.

File: scene/main/viewport.cpp [hunk 2]
```diff
@@ -1618,7 +1618,10 @@ void Viewport::_gui_show_tooltip() {
 	gui.tooltip_popup->set_position(r.position);
 	gui.tooltip_popup->set_size(r.size);
 
-	gui.tooltip_popup->show();
+	if (window->has_focus()) {
```
Comment (akien-mga) [inline scene/main/viewport.cpp:1621]: It seems a bit wasteful to do all those calculations to finally not show the tooltip... maybe the check could be an early return at the start instead? Or will the instantiated `gui.tooltip_popup` be useful later on even if it wasn't shown?

Comment (greatmomo) [reply scene/main/viewport.cpp:1621]: Yeah, I thought it was pretty wasteful too, but every time I tried to move the definition of the window up higher in the function, Godot crashed. I'm not really sure what I'm missing...

**Comment by reduz** [comment]
I am checking other software and displaying popups on unfocused windows seems to be normal. Even if you use multiple windows in Godot you will want this to happen, so I don't think this PR is the way to go.

**Comment by greatmomo** [comment]
It is worth noting that we currently don't get any tooltips on unfocused windows. This only happens if the editor was focused and loses focus while the tooltip timer is counting. So I think what you're describing would still require a change.

**Comment by KoBeWi** [comment]
Checked how it works in Chrome. When you hover button and alt-tab, the tooltip will not appear. But then you can hover it again while the window is unfocused and the tooltip will appear normally.

Godot right now does the opposite. You can't have tooltips in unfocused window, unless you hovered something before unfocusing.

**Comment by akien-mga** [comment]
Superseded by #77933. Thanks for your contribution nevertheless!

**Pull request closed without merging**